sdk/core/azure_core_test/src/tracing.rs

```diff
@@ -225,17 +234,30 @@ pub fn check_instrumentation_result(
     }
 }
 
+#[derive(Debug)]
+pub struct ExpectedSpanInformation<'a> {
+    pub span_name: &'a str,
+    pub status: SpanStatus,
+    pub kind: SpanKind,
+    pub attributes: Vec<(&'a str, AttributeValue)>,
+}
+
 fn check_span_information(span: &Arc<MockSpan>, expected: &ExpectedSpanInformation<'_>) {
     assert_eq!(span.name, expected.span_name);
     assert_eq!(span.kind, expected.kind);
     assert_eq!(*span.state.lock().unwrap(), expected.status);
     let attributes = span.attributes.lock().unwrap();
+    println!("Expected attributes: {:?}", expected.attributes);
+    println!("Found attributes: {:?}", attributes);
     for (index, attr) in attributes.iter().enumerate() {
         println!("Attribute {}: {} = {:?}", index, attr.key, attr.value);
         let mut found = false;
         for (key, value) in &expected.attributes {
             if attr.key == *key {
-                assert_eq!(attr.value, *value, "Attribute mismatch for key: {}", key);
+                // Skip checking the value for "<ANY>" as it is a placeholder
+                if *value != AttributeValue::String("<ANY>".into()) {
```

Comment:
This seems unwieldy. Not test authors have to know about some sentinel value for "<ANY>" to satisfy an assert? Makes me thinking passing the `InstrumentationInformation` to be asserted in whatever way feels appropriate is the right approach even more. As a test author, I probably only want to check a few things. Like, I don't care that the `http.method` attribute is `GET` (or something like that). Don't make me inadvertently assert things I don't need to.